### DISCUSSION TITLE
AttributeError: module 'appeears' has no attribute 'AppeearsDownloader'

### DISCUSSION THREAD

**Discussion by annieM459** [earthlab-education/Earth-Analytics-2023-01-Intro#375] [Q&A04 Raster Data] [answered]
Hi all! 

I am running into this error when I try to download NDVI files. It may have to do with how I had to download the appeears package in to  ([see discussion here](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/discussions/373)) Python but I can't quite figure it out. 
<img width="598" alt="Screenshot 2023-09-26 at 1 10 04 PM" src="https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/140002702/31a19f7c-73cb-46f5-95ea-56eed40f007c">

**Comment by JKinsler** [comment]
I'm running into the same issue. 

@ellenalamont17 and @nkorinek could you please share the specific version and documentation link for the appeears python module that we're supposed to be using? I'm wondering if this is a versioning or module naming issue.

When I run the command `!pip show appeears`, I get the following output. I'm running from my local environment and not the github workspace.
`Name: appeears
Version: 0.0.3
Summary: Simple CLI for NASA AppEEARS API
Home-page: https://github.com/samapriya/appeears
Author: Samapriya Roy
Author-email: samapriya.roy@gmail.com
License: Apache 2.0
Location: [/Users/johannakinsler/anaconda3/envs/jup_env/lib/python3.10/site-packages](https://file+.vscode-resource.vscode-cdn.net/Users/johannakinsler/anaconda3/envs/jup_env/lib/python3.10/site-packages)
Requires: beautifulsoup4, natsort, PyGeoj, requests, tabulate, tqdm
Required-by:`

I think the version I'm using is meant to be used a CLI (command line interface), and not a python module. See the documentation: https://github.com/samapriya/appeears

**Reply by eculler** [reply, marked as answer]
Hi all, we are not using the appeears cli tools. I have written an appeears module in out earthpy package. If you are using Codespaces it should be installed and imported as etapp. If not, you will need to install it using `pip install git+https://github.com/earthlab/earthpy@apppears`. You will also likely need the geoviews package for this assignment.

Alternatively, you can use our docker container earthlab/earth-analytics-dev-env.

Note that if you are able to import earthpy.appeears but are getting an error about the keyring, you have an old version and should rebuild your Codespace.

**Reply by eculler** [reply]
In general if you have environment questions, I need the following information as a starting point to effectively debug:

- Your operating system or platform
- Your code editor
- The output of `conda list`
- The full error message or unexpected behavior
- A minimal reproducible example of the code that reproduces the error (including all import statements)

If you are using Codespaces, I need to also see your .devcontainer.json file.

Environment problems can eat up a lot of everyone's time, so I appreciate your efforts to make debugging as smooth as possible!

**Reply by JKinsler** [reply]
Thanks for your help. Using `pip install git+https://github.com/earthlab/earthpy@apppears` worked.
